### PR TITLE
Add CancellationToken support

### LINKED ISSUE #2024: Async operations on FhirClient to take a CancellationToken that is can pass to any web requests it makes
**Is your feature request related to a problem? Please describe.**
When performing async requests the ability to provide a cancellation token will assist in permitting cancelling the requests - such as those to a web api - when a call is cancelled.
Very useful in terminology validations or resolving resources which may be cancelled too.

**Describe the solution you'd like**
Include the CancellationToken to any async FhirClient calls, and use that on any web/directory async calls it makes.

**Additional context**
Server validations, client validations, or any other async general requests performed.
I would like this on the ResourceResolver interfaces too, but can do that seperately.

### PR DESCRIPTION
**Pull request by ewoutkramer**
Added a CancellationToken parameter to all FhirClient operations.

Obsoleted non-async calls.

Fixed #2024.

### CODE REVIEW

File: src/Hl7.Fhir.STU3.Tests/Rest/FhirClientTests.cs [hunk 1]
```diff
@@ -1324,7 +1327,7 @@ private static void testHandlingHtmlErrorStatus(BaseFhirClient client)
         {
             try
             {
-                var pat = client.Read<Patient>("Patient/1");
```
Comment (brianpos) [inline src/Hl7.Fhir.STU3.Tests/Rest/FhirClientTests.cs:1327]: Not a fan of this change, if it's to make a compiler warning go away, add in a Diagnostics.Trace.WriteLine($"{pat.Id}"); or similar.
This is too useful just to put a breakpoint on what's come back from the server while diagnosing things.
Having it thrown away makes it harder to inspect things.

File: src/Hl7.Fhir.Base/Rest/HttpClientRequester.cs [hunk 2]
```diff
@@ -40,7 +41,7 @@ public HttpClientRequester(Uri baseUrl, FhirClientSettings settings, HttpClient
             _disposeHttpClient = false;
         }
 
-        public async Task<EntryResponse> ExecuteAsync(EntryRequest interaction)
+        public async Task<EntryResponse> ExecuteAsync(EntryRequest interaction, CancellationToken ct)
         {
             if (interaction == null) throw Error.ArgumentNull(nameof(interaction));
```
Comment (brianpos) [inline src/Hl7.Fhir.Base/Rest/HttpClientRequester.cs:47]: Should we add in something here to not bother calling things if the cancellation token has already fired?
(for larger batches would then prevent going into the subsequent calls once the cancel is done)

**Comment by brianpos** [comment]
This PR will address #2024

File: src/Hl7.Fhir.STU3.Tests/Rest/FhirClientTests.cs (hunk 1 above)
Comment (ewoutkramer) [reply src/Hl7.Fhir.STU3.Tests/Rest/FhirClientTests.cs:1327]: Yes, it was just to clean up the warnings. I'll adapt this change to your suggestion.

File: src/Hl7.Fhir.Base/Rest/HttpClientRequester.cs (hunk 2 above)
Comment (ewoutkramer) [reply src/Hl7.Fhir.Base/Rest/HttpClientRequester.cs:47]: I have added token.ThrowIfCancellationRequested().

**Review by mmsmits — COMMENTED** [review]
Should we have a test to check if the CancellationTokens work?

**Review by mmsmits — APPROVED** [review]

**Pull request merged**